### PR TITLE
Use dportal filepicker when on Linux

### PR DESCRIPTION
**Pull request by orowith2os**
This is my first attempt at using D, and it's pretty much just copy-pasting the appropriate parts from Creator, so it'll probably look like crap and won't work, but I'd like to help on implementing this.

### CODE REVIEW

File: source/app.d [hunk 1]
```diff
@@ -19,12 +19,14 @@ import session.framesend;
...
 
+version(linux) {
+    import dportals.filechooser;
+    import dportals.promise;
+}
 
 void main(string[] args) {
     insLogInfo("Inochi Session %s, args=%s", INS_VERSION, args[1..$]);
     
-    // Force X11 as the video driver on Linux
```
Comment (LunaTheFoxgirl) [inline source/app.d:26]: This should probably be seperate from this PR

File: source/app.d [hunk 2]
```diff
@@ -60,4 +62,156 @@ void main(string[] args) {
...
+        c_str filename = tinyfd_openFileDialog(__("Import..."), "", filters, false);
+        if (filename !is null) {
+            string file = cast(string)filename.fromStringz;
+            return file;
+        }
+        return null;
+    }
+}
+
+string incShowOpenFolderDialog(string title="Open...") {
```
Comment (LunaTheFoxgirl) [inline source/app.d:141]: Inochi Session functions should have the ins prefix.

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]

File: source/app.d (hunk 1 above)
Comment (orowith2os) [reply source/app.d:26]: oh, yeah, it should. mb.

File: source/app.d (hunk 2 above)
Comment (orowith2os) [reply source/app.d]: I believe I've changed the ones that need to be changed

File: dub.sdl [hunk 3]
```diff
@@ -56,6 +57,8 @@ configuration "full" {
 	targetType "executable"
 	versions "InBranding"
 
+	dependency "dportals" version="~>0.1.0"
```
Comment (LunaTheFoxgirl) [inline dub.sdl:60]: dportals should not be included in Windows builds.

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]

File: dub.sdl (hunk 3 above)
Comment (LunaTheFoxgirl) [reply dub.sdl:60]: wait, github just cut it off wierdly, don't mind me lol.

**Comment by orowith2os** [comment]
Merge conflicts are.... an experience

File: source/app.d [hunk 4]
```diff
@@ -58,3 +62,155 @@ void main(string[] args) {
...
+                string uri = promise.value["uris"].data.array[0].str;
+                uri = uri.replace("%20", " ");
+                return uri[7..$];
+            }
+            return null;
+        }
+    }
+}
+
+string insShowImportDialog(const(TFD_Filter)[] filters) {
```
Comment (LunaTheFoxgirl) [inline source/app.d:112]: This should probably be removed

File: source/app.d [hunk 5]
```diff
@@ -58,3 +62,155 @@ void main(string[] args) {
...
+        c_str filename = tinyfd_openFileDialog(__("Import..."), "", filters, false);
+        if (filename !is null) {
+            string file = cast(string)filename.fromStringz;
+            return file;
+        }
+        return null;
+    }
+}
+
+string insShowOpenFolderDialog(string title="Open...") {
```
Comment (LunaTheFoxgirl) [inline source/app.d:140]: And this

File: source/app.d [hunk 6]
```diff
@@ -58,3 +62,155 @@ void main(string[] args) {
...
+        c_str filename = tinyfd_openFileDialog(title.toStringz, "", filters, false);
+        if (filename !is null) {
+            string file = cast(string)filename.fromStringz;
+            return file;
+        }
+        return null;
+    }
+}
+
+string insShowSaveDialog(const(TFD_Filter)[] filters, string fname, string title = "Save...") {
```
Comment (LunaTheFoxgirl) [inline source/app.d:190]: And this

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]
Additionally to the provided things to discard, a UI element should also be added to load a model.

**Pull request closed without merging**